cfme/tests/intelligence/test_reports_with_timelines.py

```diff
@@ -1,39 +1,121 @@
+
+    Returns:
+        unbound function object for calling during the test
     """
-    pass
+    def _setup_for_reports(request, appliance, provider, path, updates, vm_crud, register_event): # noqa
+        """ This function makes copies of the default reports so that we can test the timelines
+        on the CloudIntel->Timelines page
+
+        Args:
+            request: py.test funcarg request
+            appliance: IPAppliance object
+            provider: provider object
+            path: path to the report in the tree
+            updates: updates to the default report so that it will display events
+            vm_crud: vm for Policy events
+            register_event: function for registering events
+            args: kwargs for extra_steps
+        """
+        # first perform extra steps for Policy stuff
+        if "Policy" in path:
+            generate_policy_event(request, appliance, provider, vm_crud, register_event) # noqa
+        report = appliance.collections.reports.instantiate(
+            type=path[0], subtype=path[1], menu_name=path[2]
+        )
+        # copy the default report
+        copied_report = report.copy()
+        # update the copied report
+        copied_report.update(updates)
+        # delete report at end of test
+        request.addfinalizer(copied_report.delete)
+
+        return copied_report
+
+    return _setup_for_reports
+
+
+def generate_policy_event(request, appliance, provider, vm_crud, register_event): # noqa
+    # create necessary objects
+    action = appliance.collections.actions.create(
+        fauxfactory.gen_alpha(),
+        "Tag",
+        dict(tag=("My Company Tags", "Environment", "Development")))
+    request.addfinalizer(action.delete)
+
+    policy = appliance.collections.policies.create(
+        policies.VMControlPolicy,
+        fauxfactory.gen_alpha()
+    )
+    request.addfinalizer(policy.delete)
 
+    policy.assign_events("VM Create Complete")
+    request.addfinalizer(policy.assign_events)
+    policy.assign_actions_to_event("VM Create Complete", action)
 
-@pytest.mark.manual
-@pytest.mark.tier(2)
-@pytest.mark.parametrize('report', timelines_reports)
-def test_custom_reports_with_timelines(report):
+    profile = appliance.collections.policy_profiles.create(
+        fauxfactory.gen_alpha(), policies=[policy])
+    request.addfinalizer(profile.delete)
+
+    # assign the policy profile to the provider
+    provider.assign_policy_profiles(profile.description)
+    request.addfinalizer(lambda: provider.unassign_policy_profiles(profile.description))
+
+    register_event(target_type='VmOrTemplate', target_name=vm_crud.name,
+                   event_type='vm_create')
+
+    vm_crud.create_on_provider(find_in_cfme=True)
+
+
+@pytest.mark.parametrize("path, updates", zip(PATHS, UPDATES)) # noqa
+def test_reports_with_timelines(request, appliance, provider, path, updates, setup_for_reports, # noqa
+                                vm_crud, register_event): # noqa
```

Comment:
Question:
why you need `noqa`?

Reply:
`noqa` is actually not necessary for L113, but for L114 and L115, the link bot complains because I am importing that fixture `vm_crud` from another test.

Reply:
Here the `vm_crud` used actually uses a provider module scoped fixture, my new push will include the vm_crud fixture in my file and use only a function scoped fixture.